allow alt dns names for certs

kbroughton opened the pull request:
modifies security-setup to allow passing --dns_alt_cert_names_string
in SAN format to include extra names in cert such as wildcards.

- [x] Installs cleanly on a fresh build of most recent master branch
- [x] Upgrades cleanly from the most recent release
- [ ] Updates documentation relevant to the changes

sehqlr commented:
I got the green light for fresh installs and upgrades, and you added documentation to security-setup, I'm saying that this is ready to be merged. @ryane, what milestone should this be in?

stevendborrelli commented:
@kbroughton the documentation in http://mantl.readthedocs.org/en/latest/security/security_setup.html needs to be updated. This .rst file is located in `docs/security/security_setup.rst`. 

We can't merge until docs are updated.

kbroughton commented:
Oops, missed steve's comment.  I've updated the docs with this

If you are getting X509 errors that the url requested was example.com but the certificate is for *.node.consul or a similar mis-match, you may need to add alternate SAN names when generating the certificate.  run security-setup --dns_alt_cert_names_string "alternative SAN names for cert here"

And rebased against master this morning.

siddharthist commented:
I'm making some changes soon that will obsolete the way we do this with security-setup, but we can still improve this :smile:

kbroughton commented:
The original need for this is largely obsoleted by https://github.com/CiscoCloud/mantl/pull/1463
If there are other use cases where it might come in handy, then it can be rebased on 1463.  For example, you might want to allow the container to choose to match on either mantl CA or official wildcard certs.

siddharthist commented:
Thanks @kbroughton!
